docusaurus/docs/iOS/guides/working-with-messagelist-layout.md

```diff
@@ -0,0 +1,137 @@
+The `isLastInSequence` property enables this operation when grouping messages. 
+
+```swift
+
+class NotGroupedMessageLayoutOptionsResolver: ChatMessageLayoutOptionsResolver {
+    override func optionsForMessage(at indexPath: IndexPath, in channel: _ChatChannel<NoExtraData>, with messages: AnyRandomAccessCollection<_ChatMessage<NoExtraData>>) -> ChatMessageLayoutOptions {
+        // Get options for the message at given indexPath to change it.
+        var options = super.optionsForMessage(at: indexPath, in: channel, with: messages)
+
+        options.insert(.continuousBubble)
+        options.insert(.timestamp)
+        options.insert(.avatar)
+        
+        // Let's add authorName to the message when it's not send by current user.
+        if !message.isSentByCurrentUser && !channel.isDirectMessageChannel {
+            options.insert(.authorName)
+        }
+        
+        return options
+    }
+}
+``` 
+
+| Message grouped | Messages separated |
+| ------------- | ------------- |
+| ![Chat with default message alignment](../assets/message-layout-squared-grouping.png)  | ![Chat with left-only alignment](../assets/message-layout-squared-nogrouping.png)  |
+
+## Square avatars
+
+First create a subclass of `ChatAvatarView` and set it according to your needs. 
+
+```swift
+final class SquareAvatarView: ChatAvatarView {
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        imageView.layer.cornerRadius = 3
+    }
+}
+``` 
+
+Next, you need to set this custom view to `Components` somewhere where your customisation takes place. 
+
+```swift
+Components.default.avatarView = SquareAvatarView.self
+```
+
+| Default avatars | Square avatars |
+| ------------- | ------------- |
+| ![Chat with default message alignment](../assets/message-layout-default.png)  | ![Chat with square avatart](../assets/message-layout-squared-avatar.png)  |
+
+ 
+## Moving components of Messages in the layout
+
+To change the message layout, you need to create a subclass subclass of [`ChatMessageContentView` ](../ReferenceDocs/Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageBubbleView)
+It's important to understand the structure of `ChatMessageContentView` so you can move around views, add your custom ones or creating complex layouts.
+
+###  Custom layout 
+
+`ChatMessageContent`
+
+![](../assets/messagelist-layout-annotation.png)
+
+- `bubbleThreadMetaContainer` is a vertical container which holds `bubbleView` at the top and `metadataContainer` at the bottom by default. You can easily switch the positions for those elements or even add your own according to your needs.
+- `metadataContainer` is a horizontal container which holds  `authorNameLabel` , `timestampLabel` and `onlyVisibleForYouLabel`. 
+- `bubbleView`  is view that embeds inside `bubbleContentContainer` which is responsible for displaying `quotedMessageView` and `textView`
```

Comment:
Is it too much headache to get a visual view for this? Even a manual one, Paint style, would be super helpful

Reply:
![](../assets/messagelist-layout-annotation.png) > 
![messagelist-layout-annotation](https://user-images.githubusercontent.com/17381941/122197165-36857980-ce98-11eb-90da-be2d4d74d934.png)